Makefile

```diff
@@ -9,34 +10,41 @@ lektor/admin/node_modules: lektor/admin/package-lock.json
 	@cd lektor/admin; npm install
 	@touch -m lektor/admin/node_modules
 
-pex:
-	virtualenv pex-build-cache
-	pex-build-cache/bin/pip install --upgrade pip
-	pex-build-cache/bin/pip install pex requests wheel
-	pex-build-cache/bin/pip wheel -w pex-build-cache/wheelhouse .
-	pex-build-cache/bin/pex \
-		-v -o lektor.pex -e lektor.cli:cli \
-		-f pex-build-cache/wheelhouse \
-		--disable-cache \
-		--not-zip-safe Lektor
-	rm -rf pex-build-cache
-
+# Run tests on Python files.
 test-python:
 	@echo "---> running python tests"
-	pylint lektor
-	pytest . --tb=long -vv --cov=lektor
-
-coverage-python: test-python
-	coverage xml
+	tox -e lint
+	tox -e coverage
 
-test-js: build-js
+# Run tests on the Frontend code.
+test-js: lektor/admin/node_modules
 	@echo "---> running javascript tests"
-	@cd lektor/admin; npm run lint
+	@cd lektor/admin; npx tsc
 	@cd lektor/admin; npm test
 
-coverage-js: test-js
-	@cd lektor/admin; npm run report-coverage
-
-test: test-python test-js
-
-coverage: coverage-python coverage-js
+.PHONY: lint
+# Lint code.
+lint:
+	pre-commit run -a
+	tox -e lint
+
+.PHONY: test
+test: lint test-python test-js
+
+.PHONY: test-all
+# Run tests on all supported Python versions.
+test-all: lint test-js
```

Comment:
The normal way to "test all" in a python project which uses `tox` is simply to run `tox` (with no args).  (That runs all the environments listed in the `[tox]envlist` setting.)  I think having to run `make test-all` to do this is fairly counter-intuitive.

I can see arguments both ways as to whether tox should be used to run the JS tests (though I lean towards using tox for those, too) but I think `tox.ini` should be the canonical "top-level" test configuration for the python tests, at least.  (I.e. just running `tox` should run, essentially, all the tests — or at least all the python tests — that are run by CI.)

Reply:
I am aware of that. However, since make acts as a the top level test entry here (and I don't think it makes sense to test all Python versions by default), I've added it here. I can also add the Python part to tox.